Do not double escape Operator in smarty secure mode

eileenmcnaughton opened the pull request:
Overview
----------------------------------------
Do not double escape Operator in smarty secure mode

Before
----------------------------------------

![image](https://user-images.githubusercontent.com/336308/206937916-992f085a-3364-4b69-9adc-f09090a54de7.png)


After
----------------------------------------
Erm - better but not still flawed

![image](https://user-images.githubusercontent.com/336308/206938029-c8b4c7d1-ed43-4c33-9fa8-717eace66db1.png)


![image](https://user-images.githubusercontent.com/336308/206937998-6c8ed532-bf34-424b-8f1f-5088a45f2014.png)


Technical Details
----------------------------------------
We already have html escaping on this field - it's enough

Comments
----------------------------------------

eileenmcnaughton committed:
Remove no-longer-used noisily deprecated function

eileenmcnaughton committed:
Do not double escape Operator in smarty secure mode

eileenmcnaughton commented:
I'm feeling a bit gaslit - still seeing this complie as 
```
    <?php else: ?>
        <?php $_from = ((is_array($_tmp=$this->_tpl_vars['orClauses']))
        ? $this->_run_mod_handler('escape', true, $_tmp, 'htmlall') : CRM_Core_Smarty::escape($_tmp, 'htmlall'));
        if (!is_array($_from) && !is_object($_from)) { settype($_from, 'array'); }$this->_foreach['criteria'] = array('total' => count($_from), 'iteration' => 0);
if ($this->_foreach['criteria']['total'] > 0):
```

despite the `smarty:nodefaults`
https://www.smarty.net/docsv2/en/variable.default.modifiers.tpl

seamuslee001 commented on templates/CRM/common/displaySearchCriteria.tpl:
```diff
@@ -36,10 +36,10 @@
             {if !$smarty.foreach.criteria.last}
                 {if !empty($operator)}
-                  <span class="font-italic">...{$operator|escape}...</span>
+                  <span class="font-italic">...{$operator|smarty:nodefaults|escape:html}...</span>
```
I don't think you need to do the :html because I think that is the default mode with the escape function iirc

yashodha committed:
Merge pull request #25152 from eileenmcnaughton/create_act

Remove no-longer-used noisily deprecated function `CRM_Contact_Form_Task_PDFLetterCommon::createActivities`

larssandergreen committed:
allow hyphens in tracked URLs in text emails

eileenmcnaughton commented:
@demeritcowboy shame you couldn't prove I'm not crazy - that would have been nice if you could have

demeritcowboy commented:
I'll see what I can do for xmas. The internet must have a fine selection of "you are not crazy" gifts. Supply and demand always wins, and probably lots of demand.

As long as a captcha doesn't require proving I'm not crazy first.

yashodha committed:
Merge pull request #25149 from larssandergreen/allow-dashes-in-tracked-URLs-for-text-in-flexmailer

Flexmailer: Prevent broken urls containing hyphens when click tracking is enabled for plain text mailings

totten committed:
getSiteStats - Use consistent report format

When general usage stats are enabled, the format should be consistent across
versions.

Before (4.7 - 5.56): Report stat as 'Delivered'
Before (5.57.beta1): Report stat as 'MailingEventDelivered'
After (5.57.beta1): Report stat as 'Delivered'

demeritcowboy committed:
Merge pull request #25157 from civicrm/5.57

5.57

demeritcowboy committed:
Merge pull request #25158 from totten/5.57-entity-stat

getSiteStats - Use consistent report format

eileenmcnaughton committed:
Merge pull request #25163 from civicrm/5.57

5.57

eileenmcnaughton committed:
Fix smarty notices on Open Case

seamuslee001 committed:
Merge pull request #25165 from eileenmcnaughton/smarty_case

Fix smarty notices on Open Case

demeritcowboy commented on templates/CRM/common/displaySearchCriteria.tpl:
```diff
@@ -36,10 +36,10 @@
         {foreach from=$orClauses name=criteria item=item}
             <div class="qill">
-            {$item|escape}
+            {$item|smarty:nodefaults|escape}
```
I see it - also need to put nodefaults on line 37.

eileenmcnaughton commented:
@demeritcowboy I tried adding the extra nodefaults but I think the issue is this

- with smarty defaults enabled it escapes each item in an array
- the nodefaults on the array escaping is maybe not working?
- it then escapes each item individually - we could use nodefautls here maybe - but not if we have hard-coded in to do some escaping

I think @totten might have proposed something that does some more involved inspection to find & handle this situation (pretty sure Tim worked something up when I was focussed on getting a larger issue resolved & didn't have the headspace to look at it while the message template stuff was not merged)

eileenmcnaughton commented:
I don't think this is 'review-ready' - it seems to need another round of work - so I'll close it for now & track in https://lab.civicrm.org/dev/core/-/issues/4077

Also it's pointless cos @demeritcowboy couldn't prove I'm not crazy.....

demeritcowboy commented:
It's on my todo list to prove but I got stuck on the unit test.

eileenmcnaughton commented:
@demeritcowboy Schrodingers sanity?

demeritcowboy commented:
How many cats do you have? That might make it harder to prove...
